### PR TITLE
Enabled spotless for enrich gradle project.

### PR DESCRIPTION
**Pull request by martijnvg**
The enrich project doesn't have much history as all the other gradle projects,
so it makes sense to enable spotless for this gradle project.

cc: @mark-vieira @pugnascotia

### CODE REVIEW

File: x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/AbstractEnrichProcessor.java [hunk 1]
```diff
@@ -155,13 +171,12 @@ int getMaxMatches() {
...
         return (req, handler) -> {
-            client.execute(EnrichCoordinatorProxyAction.INSTANCE, req, ActionListener.wrap(
-                resp -> {
-                    handler.accept(resp, null);
-                },
-                e -> {
-                    handler.accept(null, e);
-                }));
+            client
+                .execute(
```
Comment (martijnvg) [inline x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/AbstractEnrichProcessor.java]: I'm not so happy that `.execute(` is on a new line. This happens also a lot in other places.

File: x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/AbstractEnrichProcessor.java [hunk 2]
```diff
@@ -35,15 +35,31 @@
     protected final String matchField;
     protected final int maxMatches;
 
-    protected AbstractEnrichProcessor(String tag, Client client, String policyName, String field, String targetField,
```
Comment (martijnvg) [inline x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/AbstractEnrichProcessor.java]: I prefer not putting all parameters on a new line and only adding a new line when it reaches the line length limit.

File: x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/AbstractEnrichProcessor.java (hunk 1 above)
Comment (pugnascotia) [reply x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/AbstractEnrichProcessor.java]: I've gone backwards and forwards on this setting ("Qualified invocations") a few times. I think we should actually change it back to what I originally used, which was "Wrap all elements, except first element if not necessary". It doesn't work in every situation, but it's probably better on balance.

You can change this setting with the following patch. Would you mind applying it?

```patch
diff --git a/.eclipseformat.xml b/.eclipseformat.xml
index 9ff0aa1f408..ebda0af1503 100644
--- a/.eclipseformat.xml
+++ b/.eclipseformat.xml
@@ -205,7 +205,7 @@
         <setting id="org.eclipse.jdt.core.formatter.alignment_for_resources_in_try" value="48"/>
         <setting id="org.eclipse.jdt.core.formatter.use_tabs_only_for_leading_indentations" value="false"/>
         <setting id="org.eclipse.jdt.core.formatter.parentheses_positions_in_try_clause" value="separate_lines_if_wrapped"/>
-        <setting id="org.eclipse.jdt.core.formatter.alignment_for_selector_in_method_invocation" value="48"/>
+        <setting id="org.eclipse.jdt.core.formatter.alignment_for_selector_in_method_invocation" value="80"/>
         <setting id="org.eclipse.jdt.core.formatter.never_indent_block_comments_on_first_column" value="false"/>
         <setting id="org.eclipse.jdt.core.formatter.keep_code_block_on_one_line" value="one_line_if_empty"/>
         <setting id="org.eclipse.jdt.core.formatter.insert_space_after_opening_paren_in_synchronized" value="do not insert"/>
```

File: x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/AbstractEnrichProcessor.java (hunk 2 above)
Comment (pugnascotia) [reply x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/AbstractEnrichProcessor.java]: For background on why it's configured this way: mostly I was guided by what Prettier (https://prettier.io/) does, whose strategy is broadly to try to fit everything on one like, but when you’re forced to wrap then to put most things on their own lines. I’m fairly sure that Prettier stems from [Wadler’s algorithm from “A prettier printer”](https://homepages.inf.ed.ac.uk/wadler/papers/prettier/prettier.pdf
) but I haven’t read the paper enough to confidently state that.

Overally I was going for consistency - for example, in this file, there were `List.of` calls which spanned multiple lines even before formatting, and they have one argument per line. So having one rule for method arguments and one for other types of expressions isn't especially consistent.

File: x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/AbstractEnrichProcessor.java (hunk 1 above)
Comment (martijnvg) [reply x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/AbstractEnrichProcessor.java]: Thanks, I've changed that setting via: 910a5ba9f657b3c9a9a9cefde4729fb462dce2a9 
This is looks better now

File: x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/AbstractEnrichProcessor.java (hunk 2 above)
Comment (martijnvg) [reply x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/AbstractEnrichProcessor.java]: Thanks for sharing the background here. I understand now why and I think it is ok.
Also I didn't feel strongly about this comment than the first comment.

**Review by pugnascotia — APPROVED** [review]

**Comment by martijnvg** [comment]
Thanks @pugnascotia!

**Review by alpar-t — APPROVED** [review]
LGTM 
Thanks for picking this up!

**Review by mark-vieira — APPROVED** [review]
LGTM.

Let's just make sure this gets backported to any branches that also include enrich to save us the pain of future backports. Not sure if this only exists in `8.0`.

**Comment by martijnvg** [comment]
> Let's just make sure this gets backported to any branches that also include enrich to save us the pain of future backports.

Spotless doesn't exists in the 7.x and 7.5 branches, so I will backport the code changes to 7.x and 7.5. Unfortunately these can't be applied cleanly, because there are some difference (e.g. List.of(...) vs. Arrays.asList(...)).

**Comment by atorok** [comment]
@pugnascotia in light of this development, I think we should backport spotless to these older versions

**Comment by pugnascotia** [comment]
I was sure I’d already backported it. I’ll check.

**Comment by martijnvg** [comment]
> I was sure I’d already backported it. I’ll check.

I commented too soon.. I was just running the `spotlessApply` command on 7.x, but obviously that doesn't work yet, because it hasn't been enabled in any gradle project. Need more coffee...

**Pull request merged**